test/spec/controllers/eventForm.spec.js

```diff
@@ -0,0 +1,309 @@
+      );
+
+      expect(GrowlService.growl).toHaveBeenCalledTimes(1);
+      expect(GrowlService.growl.calls.argsFor(0)[3]).toBe(
+        'Page "Page 2" has been deleted.',
+      );
+    });
+  });
+
+  it('copyBlock copies an existing block', () => {
+    const existingBlock = scope.conference.registrationPages[1].blocks[3];
+    scope.copyBlock(existingBlock.id);
+    const newBlock = scope.conference.registrationPages[1].blocks[4];
+
+    expect(newBlock.id).not.toBe(existingBlock.id);
+    expect(newBlock.position).toBe(4);
+    expect(newBlock.title).toBe('Dropdown Question (copy)');
+    expect(newBlock.rules[0].id).not.toBe(existingBlock.rules[0].id);
+    expect(newBlock.rules[0].blockId).toBe(newBlock.id);
+  });
+
+  describe('insertBlock', () => {
+    it('adds a new block without a default profile', () => {
+      const page = scope.conference.registrationPages[0];
+      const previousFirstBlock = page.blocks[0];
+      scope.insertBlock(
+        'nameQuestion',
+        page.id,
+        0,
+        'Name',
+        undefined,
+        undefined,
+      );
+      const newBlock = page.blocks[0];
+
+      expect(newBlock.pageId).toBe(page.id);
+      expect(newBlock.title).toBe('Name');
+      expect(page.blocks[1]).toBe(previousFirstBlock);
+    });
+
+    it('adds a new block with an unused default profile', () => {
+      const page = scope.conference.registrationPages[0];
+      scope.insertBlock(
+        'phoneQuestion',
+        page.id,
+        0,
+        'Telephone',
+        'PHONE',
+        undefined,
+      );
+
+      expect(page.blocks[0].profileType).toBe('PHONE');
+    });
+
+    it('adds a new block with a used default profile', () => {
+      const page = scope.conference.registrationPages[0];
+      scope.insertBlock(
+        'addressQuestion',
+        page.id,
+        0,
+        'Address',
+        'ADDRESS',
+        undefined,
+      );
+
+      expect(page.blocks[0].profileType).toBe(null);
+    });
+  });
+
+  describe('deleteBlock', () => {
+    it('deletes a block', () => {
+      spyOn(GrowlService, 'growl');
+
+      scope.deleteBlock('0b876382-5fd1-46af-b778-10fc9b1b530d', true);
+
+      expect(GrowlService.growl).toHaveBeenCalledTimes(1);
+      expect(GrowlService.growl.calls.argsFor(0)[3]).toBe(
+        '"Multiple Choice Question" has been deleted.',
+      );
+    });
+
+    it('refuses to delete a block with dependent rules', () => {
+      spyOn(modalMessage, 'error');
+      scope.conference.registrationPages.push(testData.rulesPage);
+
+      scope.deleteBlock('0b876382-5fd1-46af-b778-10fc9b1b530d', true);
+
+      expect(modalMessage.error).toHaveBeenCalledTimes(1);
+      const errorMessage = modalMessage.error.calls.argsFor(0)[0].message;
+
+      expect(errorMessage).toContain('<li>Question</li>');
+    });
+  });
+
+  it('addNewPage adds a new page', () => {
+    scope.addNewPage();
```

Comment:
Should you add a expect the pages to have a legnth of 2 first, so we know a page was added.